include/deal.II/base/config.h.in

```diff
@@ -150,6 +150,18 @@
 #cmakedefine DEAL_II_HAVE_CXX17
 #cmakedefine DEAL_II_HAVE_CXX20
 
+/**
+ * If we have C++20 available, we can have concepts and requires
```

Comment:
We should probably check that we define DEAL_II_HAVE_CXX20 only if the compiler supports concepts.

Reply:
Yes, adding something using concepts in cmake/checks/check_01_cxx_features.cmake.